internal/core/metadata/rest_provisionwatcher.go

```diff
@@ -34,8 +34,8 @@ func restGetProvisionWatchers(w http.ResponseWriter, _ *http.Request) {
 	}
 
 	// Check the length
-	if len(res) > Configuration.Service.MaxResultCount {
-		httpErrorHandler.Handle(w, errors.New("Max limit exceeded"), errorconcept.Default.RequestEntityTooLarge)
+	if err = checkMaxLimit(Configuration.Service.MaxResultCount); err != nil {
```

Comment:
As suggested in my other comment can we verify the argument being passed is correct?